utilities/transactions/write_prepared_txn_db.h

```diff
@@ -778,12 +812,28 @@ class AddPreparedCallback : public PreReleaseCallback {
   }
   virtual Status Callback(SequenceNumber prepare_seq,
                           bool is_mem_disabled __attribute__((__unused__)),
-                          uint64_t log_number) override {
+                          uint64_t log_number, size_t index,
+                          size_t total) override {
+    assert(index < total);
+    // To reduce lock intention with the concurrent prepare requests, lock on
```

Comment:
Lock contention?

Also, I don't know if I'd call it lock contention because it looks like you actually increase lock contention by holding the lock for longer periods of time (whereas before, it was shorter, but more frequent). I think you're just saving CPU?

Reply:
Perhaps I can say "reduce lock acquisition cost".